CDCgov/prime-reportstream
# Remove dummy action from build_hub

This PR removes the "dummy" action from the Build Hub workflow. 

Test Steps:
1. Open up a PR with changes to one or more of the following paths:
            - 'prime-router/**'
            - 'operations/**'
            - '.github/actions/build-backend/action.yml'
            - '.github/actions/deploy-backend/action.yml'
            - '.github/workflows/build_hub.yml'
   Build Hub should run as part of the checks and actually build the router.
2. Open up a PR *without* any changes to the paths above. Build Hub should still run, but quicker: should skip building the router.

## Changes
- Removed `build_hub_dummy.yml` and changed build_hub.yml to _always_ kick off but conditionally skip the router build based on which files actually changed.

## Checklist

### Testing
## Linked Issues
- Fixes #12235
